src/nile/core/types/transactions.py

```diff
@@ -47,7 +47,7 @@ class Transaction(ABC):
     """
 
     account_address: int = 0
-    max_fee: int = 0
+    max_fee: Union[int, None] = 0
```

Comment:
```suggestion
    max_fee: int = 0
```

This will set `max_fee` as 0 even when None is passed (max_fee=None). We don't need to normalize the max_fee in `post_init`.

Reply:
>  If the intention is to support max_fee=max_fee with max_fee equal None from the account methods

That indeed is the intention

> I think is more clear to change the call in the account to max_fee=max_fee or 0 instead.

I agree it's more clear. While this avoids the need to normalize the `max_fee` in `Transaction`, we'd need to create a new variable to save the original `max_fee` passed to the account method (to check if it's None because that's the trigger for the fee estimation).

If you think either way is too much to accommodate the 0 `max_fee` edge case, I can apply the suggestions and keep it simple

Reply:
We need an extra variable? I'm thinking in something like this (for each method):

```
        # Create the transaction
        transaction = DeclareTransaction(
            account_address=self.address,
            contract_to_submit=contract_name,
            max_fee=max_fee or 0,
            nonce=nonce,
            network=self.network,
            overriding_path=overriding_path,
        )

        tx_wrapper = DeclareTxWrapper(
            tx=transaction,
            account=self,
            alias=alias,
        )

        await _set_estimated_fee_if_none(max_fee, tx_wrapper)
        return tx_wrapper
```

The change is just `max_fee=max_fee or 0` in the `DeclareTransaction` call.

`_set_estimated_fee_if_none` would still receive None. The transaction doesn't have to, right?

Reply:
Ahhh directly in the function call. Yeah, that's much cleaner. Thank you, sir! Will update